veecle-osal-embassy/src/thread.rs

```diff
@@ -12,28 +12,28 @@ pub struct Thread;
 impl ThreadAbstraction for Thread {
     #[cfg(not(target_os = "none"))]
     fn current_thread_id() -> u64 {
-        use std::cell::Cell;
+        use std::cell::OnceCell;
         use std::sync::atomic::{AtomicU64, Ordering};
 
         /// Global counter for generating unique thread ids.
         static NEXT_THREAD_ID: AtomicU64 = AtomicU64::new(1);
 
         std::thread_local! {
             /// Thread-local storage for the current thread's id.
-            static THREAD_ID: Cell<u64> = const { Cell::new(0) };
+            static THREAD_ID: OnceCell<u64> = const { OnceCell::new() };
         }
 
         THREAD_ID.with(|id| {
             let current = id.get();
-            if current == 0 {
-                // `Relaxed` here is enough because we don't care about what values various threads
-                // see, just that they're unique (assuming that creating 2^64 threads is impractical
-                // so overflow can't happen).
+            if let Some(current) = current {
```

Comment:
With `OnceCell` this can be `get_or_init`

Reply:
I switched to `LazyCell` as that allows us to get rid of that `if` completely.